### DISCUSSION TITLE
A few ideas

### DISCUSSION THREAD

**Discussion by metaleap** [zigtools/zls#1485] [Ideas]
Heya, new fan here, ZLS is looking real good in action!

I installed it yesterday and used its "Install Zig" feature to install 0.11.0, didn't want "Nightly". I love this feature of an extension-managed download-install-update lifecycle. Well, a message that keeps popping up today (but not yesterday) is:

> Zig `0.11.0` is older than ZLS `0.12.0-dev.105+5405178`. Update Zig to avoid unexpected behavior.

**Idea:** since ZLS offers this extension-managed Zig-keeping, it could offer an option (command or button-on-the-above-quoted-prompt) to install _exactly the very version_ it so suggests (since the offered "Nightly" might be already newer and thus _also_ risk "unexpected behavior"). (In case that has binaries stashed somewhere of course, maybe non-0.x.0 versions don't?)

With that out the way, some more sideshow stuff:

- In addition to `zig.buildOnSave`, an optional (default `false`) `zig.testOnSave` would be something real neat. Might try to devise a PR on that eventually unless someone else is already itching to cook that up now, or unless that somehow goes against, dunno, scope/philosophy or sth  =)  of course, tests can have the edge-case of using `stdin` so I guess such a feature would omit the `zig test` call for such files on principle.
  - since it would seem that `zig test` error-message formatting is just like the compilation one, I'd also go for diagnostics publishing into VSC's _'Problems'_ panel.

- Of course that also suggests a command / code-lens / quick-action or similar ux affordance to run test on file or package, regardless of any "on save" or not.

- Not an idea, but since I'm already typing: auto-complete seems to only work at my end when at the end of the line. Ie. it works when the line is merely `std.` or `pkg.foo.` etc, but when I try to trigger it (ctrl+space at my end) say at the first dot of `std.debug.print`, then _'No suggestions.'_ This might be a known Issue? If so I haven't found it at a glance... guess the ugly-quick'n'dirty fix would be for the autocomplete handler to only consider what's to the left of the caret (since that works, ie when I  manually cut out what's to the right of the caret)  :grin:

**Comment by Vexu** [comment]
I'm working on being able to install specific version of ZLS in https://github.com/ziglang/vscode-zig/pull/138 but I need Auguste to update the release tarballs and he's been on a work trip.

**Comment by SuperAuguste** [comment]
Back from the work trip @Vexu, working on this now :)
